odoo/addons/base/models/ir_cron.py

```diff
@@ -516,21 +515,20 @@ def _update_failure_count(self, job, status):
 
     def _clear_schedule(self, job):
         """Remove triggers for the given job."""
+        now = self.env.cr.now().replace(microsecond=0)
         self.env.cr.execute("""
             DELETE FROM ir_cron_trigger
             WHERE cron_id = %s
-              AND call_at <= (now() at time zone 'UTC')
-        """, [job['id']])
+              AND call_at <= %s
+        """, [job['id'], now])
 
     def _reschedule_later(self, job):
         """
         Reschedule the job to be executed later, after its regular
         interval or upon a trigger.
         """
-        # Use the user's timezone to compare and compute datetimes, otherwise unexpected results may appear.
-        # For instance, adding 1 month in UTC to July 1st at midnight in GMT+2 gives July 30 instead of August 1st!
-        now = fields.Datetime.context_timestamp(self, datetime.utcnow())
-        nextcall = fields.Datetime.context_timestamp(self, job['nextcall'])
+        now = self.env.cr.now().replace(microsecond=0)
```

Comment:
The comment clearly states that we want to use the user's timezone.

Reply:
The comment is wrong. In general, at the server-side, you want to avoid localized datetime.
All localized datetimes are removed, so it is much simpler to reason about.

Reply:
Added back when adding the interval, rephrased the comment to indicate why.